src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -428,6 +428,10 @@ private async Task<NegotiationResponse> NegotiateAsync(Uri url, HttpClient httpC
                         {
                             negotiateResponse = NegotiateProtocol.ParseResponse(responseStream);
                         }
+                        if (!string.IsNullOrEmpty(negotiateResponse.Error))
+                        {
+                            throw new InvalidOperationException(negotiateResponse.Error);
```

Comment:
`InvalidOperationException` isn't really right here... We don't know anything about the underlying problem, it's no better than just throwing `Exception`.